address technical feedback on API caching blog post (#1206)
temporalio/documentation: blog/2022-05-26-caching-api-requests-with-long-lived-workflows.md

  startToCloseTimeout: '1 minute',
});

type ExchangeRatesType = { [key: string]: number };

export const getExchangeRatesQuery = defineQuery<ExchangeRatesType | undefined, [string]>('getExchangeRates');

export async function exchangeRatesWorkflow(): Promise<void> {

>   let rates: ExchangeRatesType | undefined = null;

Reviewer: Did you run this code throw a TS compiler?

```suggestion
  let rates: ExchangeRatesType | undefined = undefined;
```

    const tomorrow = new Date(today);
    tomorrow.setHours(12, 0, 0, 0);
    tomorrow.setDate(tomorrow.getDate() + 1);

>     await sleep(tomorrow.valueOf() - today.valueOf());

Reviewer: Also could abstract this to some helper function that doesn't need to be implemented for this post

}
```

Following is the output of `start-workflow.ts`.

```
Started workflow exchange-rates-workflow
```

> Once you have started the Workflow, you can execute a Query to get the latest exchange rates using the following `query-workflow.ts` script.

Reviewer 2: ```suggestion
After you start the Workflow, you can execute a Query to get the latest exchange rates using the following `query-workflow.ts` script.
```

```ts
import { WorkflowClient } from '@temporalio/client';
import { getExchangeRatesQuery } from '../workflows';
import { toYYYYMMDD } from '../shared';

const date: string = process.argv[2] || toYYYYMMDD(new Date()); 

run().catch((err) => {
  console.error(err);
  process.exit(1);

> });

Reviewer: nit: with ESM you can use top level awaits and get rid of this `run` wrapper

The `exchangeRatesWorkflow` can run for an unlimited period of time: days, months, even years.
However, Temporal caps a Workflow at 50,000 events. (See the [Time constraints](https://docs.temporal.io/workflows#time-constraints) section in [Temporal Workflows](https://docs.temporal.io/workflows).)

> In the `exchangeRatesWorkflow`, eight events are fired at every iteration of the `while` loop, assuming no API errors.

Reviewer 2: ```suggestion
In the `exchangeRatesWorkflow`, eight events are fired during each iteration of the `while` loop, assuming no API errors.
```

1. `EVENT_TYPE_TIMER_FIRED`: the `setTimeout()` resolved and it's time to refresh the exchange rates
2. `EVENT_TYPE_ACTIVITY_TASK_SCHEDULED`: the Temporal server scheduled the `getExchangeRates()` activity
3. `EVENT_TYPE_ACTIVITY_TASK_STARTED`: the `getExchangeRates()` activity started executing
4. `EVENT_TYPE_ACTIVITY_TASK_COMPLETED`: the `getExchangeRates()` activity completed successfully
5. `EVENT_TYPE_WORKFLOW_TASK_SCHEDULED`: the Temporal server scheduled the Workflow logic that handles `getExchangeRates()`
6. `EVENT_TYPE_WORKFLOW_TASK_STARTED`: the Workflow resumed
7. `EVENT_TYPE_WORKFLOW_TASK_COMPLETED`: the Workflow paused
8. `EVENT_TYPE_TIMER_STARTED`: the Workflow used `setTimeout()` to pause until tomorrow

> With one API request per day, the `exchangeRatesWorkflow()` can run for almost 6,250 days (approximately 17 years) before running into the 50,000 event limit.

Reviewer 2: ```suggestion
With one API request per day, `exchangeRatesWorkflow()` can run for almost 6,250 days (approximately 17 years) before running into the 50,000 event limit.
```

> With one API request per day, the `exchangeRatesWorkflow()` can run for almost 6,250 days (approximately 17 years) before running into the 50,000 event limit.

Reviewer 3: My recent understanding is that in reality we want to keep it under 500.

Reviewer 4: Event history length under 500? Due to replay time?
